### DISCUSSION TITLE
Turn off "Added support for autogenerate comparison of indexes on PostgreSQL which include SQL expressions"

### DISCUSSION THREAD

**Discussion by AntonOfTheWoods** [sqlalchemy/alembic#1210] [Usage Questions]
From what I understood at the time, previously if I needed a function (lowercase) index with Postgres, I needed to add that manually. I did that in a revision and it worked, with warnings like `UserWarning: Skipping reflected expression-based index 'ix_zhsubtlexlookup_source_text_lower'`. Then I upgraded to alembic 1.10.0, and it appears it now tells me it's going to remove the indexes I added manually, before simply errorring out with:

```
sqlalchemy.exc.ArgumentError: Index 'ix_zhsubtlexlookup_source_text_lower' is against table 'zhsubtlexlookup', and cannot be associated with table 'zhsubtlexlookup'.
```
I thought I might be able to add the indexes but it continues to seem to want to ignore such indexes (i.e, if I add the index to the table definition it still completely ignores it, and I get the same error). I am assuming it is due to the following:

https://alembic.sqlalchemy.org/en/latest/changelog.html#change-477beffd1a8eb4fb324e68813c87bb7f

Am I correct in assuming that now alembic continues to not properly support function indexes on Postgres but doesn't allow having them done manually either? Is there any way around this apart from sticking to 1.9.4?

**Comment by CaselIT** [comment]
Hi,

Things should work correctly, the idea is that alembic does support function indexes, like any other index.
What version of sqlalchemy and alembic are you using?
Can you provide examples of the indexes you are trying do add?

**Reply by AntonOfTheWoods** [reply]
Sorry, I'm an idiot, I need to learn how to create indexes on mixin classes properly. You're right, everything works as expected. Sorry!

**Reply by CaselIT** [reply]
Great to hear that everything worked out. Let us know if you find other issues, as with all new (and old) things there may be bugs!
